# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/windows/ia2/ia2AccessibleTable.cpp
# Commit: 6d0f4821ddcf
# Full Hash: 6d0f4821ddcfca1b59e4db3befc3ec9c12335891
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-03-23 09:49:32
# Regressor Bug: 1735970
# File Overlap Count: 1
# Description:
#   Bug 1735970 part 3: Add TableAccessibleBase and TableCellAccessibleBase. r=morgan
#   
#   These are needed to support RemoteAccessible tables.
#   Stuff specific to LocalAccessible is still in TableAccessible and TableCellAccessible, which now inherit from the new Base classes.
#   
# ==============================================================================

diff -r d82660fb5408 -r 6d0f4821ddcf accessible/windows/ia2/ia2AccessibleTable.cpp
--- a/accessible/windows/ia2/ia2AccessibleTable.cpp	Wed Mar 23 04:00:25 2022 +0000
+++ b/accessible/windows/ia2/ia2AccessibleTable.cpp	Wed Mar 23 04:00:26 2022 +0000
@@ -512,7 +512,7 @@
   TableAccessible* table = TableAcc();
   if (!table) return CO_E_OBJNOTCONNECTED;
 
-  AutoTArray<LocalAccessible*, 30> cells;
+  AutoTArray<Accessible*, 30> cells;
   table->SelectedCells(&cells);
   if (cells.IsEmpty()) return S_FALSE;
 
@@ -521,8 +521,7 @@
   if (!*aCells) return E_OUTOFMEMORY;
 
   for (uint32_t i = 0; i < cells.Length(); i++) {
-    RefPtr<IAccessible> cell;
-    cells[i]->GetNativeInterface(getter_AddRefs(cell));
+    RefPtr<IAccessible> cell = MsaaAccessible::GetFrom(cells[i]);
     cell.forget(&(*aCells)[i]);
   }
 